Fixes lp#1851441: WARN vs ERROR

anastasiamac opened the pull request:
## Description of change

Some commands message warning when an error occurs. It fits some scenarios but not when a user explicitly asked for something and that something did not happen.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1851441

anastasiamac commented:
![](https://github.trello.services/images/mini-trello-icon.png) [WARN vs ERROR](https://trello.com/c/zs6pkMbX/974-warn-vs-error)

wallyworld commented on cmd/juju/cloud/regions.go:
```diff
@@ -133,21 +133,20 @@ func (c *listRegionsCommand) Run(ctxt *cmd.Context) error {
 	if c.Client {
 		if err := c.findLocalRegions(ctxt); err != nil {
-			ctxt.Warningf("%v", err)
+			ctxt.Infof("ERROR %v", err)
```
These ones should still be a warning?
The command still completes and shows some info, it's just that not everything could be retrieved and those things that couldn't are warned about.

wallyworld commented on cmd/juju/cloud/show.go:
```diff
@@ -142,12 +142,12 @@ func (c *showCloudCommand) Run(ctxt *cmd.Context) error {
 		if remoteCloud != nil {
 			if err := c.displayCloud(ctxt, remoteCloud, fmt.Sprintf("Cloud %q from controller %q:\n", c.CloudName, c.ControllerName), showRemoteConfig, remoteErr); err != nil {
-				ctxt.Warningf("%v", err)
+				ctxt.Infof("ERROR %v", err)
```
ditto

wallyworld commented on cmd/juju/cloud/showcredential.go:
```diff
@@ -91,18 +91,21 @@ func (c *showCredentialCommand) Run(ctxt *cmd.Context) error {
 		result, err := c.localCredentials(ctxt)
 		if err != nil {
-			ctxt.Infof("client credential content lookup failed: %v", err)
+			ctxt.Infof("ERROR client credential content lookup failed: %v", err)
```
ditto

wallyworld commented on cmd/juju/cloud/listcredentials.go:
```diff
@@ -252,20 +252,25 @@ func (c *listCredentialsCommand) Run(ctxt *cmd.Context) error {
 		credentials.Client, err = c.localCredentials(ctxt)
 		if err != nil {
-			ctxt.Warningf("%v", err)
+			ctxt.Infof("ERROR %v", err)
```
ditto

anastasiamac commented:
Thank you for the review \o/

We have discussed your concerns off-line and have decided to use the distinction of 'if a user asked for something and we could not get it' as a guide to determine whether to ERR or to WARN.

This will allow users to differentiate between 'a system failed to get my information' (ERROR) vs 'a system tried to get my information but there is nothing to show/list'.

We have agreed that my implementation is the way to go without addressing the comments.

anastasiamac commented:
$$merge$$

anastasiamac commented:
new and awesome way the check failed (unrelated to this change): 12:22:46 Cloning into '/home/ubuntu/go/src/golang.org/x/tools'...
12:22:46 fatal: The remote end hung up unexpectedly
12:22:46 fatal: early EOF
12:22:46 fatal: index-pack failed
12:22:46 package golang.org/x/tools/go/ast/astutil: exit status 128
12:22:47 package golang.org/x/tools/go/gcexportdata: cannot find package "golang.org/x/tools/go/gcexportdata" in any of:
12:22:47 	/snap/go/4765/src/golang.org/x/tools/go/gcexportdata (from $GOROOT)
12:22:47 	/home/ubuntu/go/src/golang.org/x/tools/go/gcexportdata (from $GOPATH)
12:22:47 + cleanup
12:22:47 + exit_code=1
